### PR TITLE
Populate data stream fields when xOpaqueId not provided

### PR DESCRIPTION
**Pull request by pgomulka**
datastream related fields should be always be provided.
message field is already set with a lazy value (Object.toString) within
ESLogMessage
Also removing test cases to directly access deprecation logger with LogManger.getLogger("deprecation"). This is disallowed. `DeprecationLogger.getLogger` should be used.

### CODE REVIEW

**Review by pugnascotia — APPROVED** [review]
These changes look good. As discussed, if we can add a test case too that would be ideal.

File: qa/logging-config/src/test/java/org/elasticsearch/common/logging/JsonLoggerTests.java [hunk 1]
```diff
@@ -203,22 +202,17 @@ public void testDeprecatedMessageWithoutXOpaqueId() throws IOException {
...
-                    hasEntry("type", "deprecation"),
-                    hasEntry("log.level", "DEPRECATION"),
-                    hasEntry("log.logger", "deprecation.test"),
-                    hasEntry("cluster.name", "elasticsearch"),
-                    hasEntry("node.name", "sample-name"),
-                    hasEntry("message", "deprecated message2"),
+                    hasEntry("data_stream.type", "logs"),
+                    hasEntry("data_stream.datatype", "deprecation"),
+                    hasEntry("data_stream.namespace", "elasticsearch"),
+                    hasEntry("ecs.version", DeprecatedMessage.ECS_VERSION),
```
Comment (pugnascotia) [inline qa/logging-config/src/test/java/org/elasticsearch/common/logging/JsonLoggerTests.java:208]: Should we check for the presence of `key` here?

Comment (pgomulka) [reply qa/logging-config/src/test/java/org/elasticsearch/common/logging/JsonLoggerTests.java:208]: yes, will add - missed it

**Comment by pgomulka** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample-windows

**Pull request merged**